design/areas.go

```diff
@@ -0,0 +1,115 @@
+package design
+
+import (
+	d "github.com/goadesign/goa/design"
+	a "github.com/goadesign/goa/design/apidsl"
+)
+
+var area = a.Type("Area", func() {
+	a.Description(`JSONAPI store for the data of a Area.  See also http://jsonapi.org/format/#document-resource-object`)
+	a.Attribute("type", d.String, func() {
+		a.Enum("areas")
+	})
+	a.Attribute("id", d.UUID, "ID of area", func() {
+		a.Example("40bbdd3d-8b5d-4fd6-ac90-7236b669af04")
+	})
+	a.Attribute("attributes", areaAttributes)
+	a.Attribute("relationships", areaRelationships)
+	a.Attribute("links", genericLinks)
+	a.Required("type", "attributes")
+})
+
+var areaAttributes = a.Type("AreaAttributes", func() {
+	a.Description(`JSONAPI store for all the "attributes" of a Area. +See also see http://jsonapi.org/format/#document-resource-object-attributes`)
+	a.Attribute("name", d.String, "The Area name", func() {
```

Comment:
As discussed, the following needs to be exposed, agnostic to how we persist in the database:

1. path ( of UUIDs upto topmost parent )
2. path_resolved ( of area_names of the above UUIDs )
3. relationship : link to all children
4. relationship : link to parent.